lib/cog/commands/bundle.ex

```diff
@@ -69,70 +75,130 @@ defmodule Cog.Commands.Bundle do
-           {:ok, bundle} ->
-             {:ok, Bundle.Status.current(bundle)}
-           {:error, :embedded_bundle}=error ->
-             error
-         end
-       {:error, _}=error ->
-         error
-     end
-   end
-   defp run(["status", bundle_name]) when is_binary(bundle_name) do
-     case find_bundle(bundle_name) do
-       {:ok, bundle} ->
-         {:ok, Bundle.Status.current(bundle)}
-       {:error, _}=error ->
-         error
-     end
-   end
-   defp run(_) do
-     {:error, :invalid_invocation}
-   end
-
-   defp action_to_status("enable"), do: :enabled
-   defp action_to_status("disable"), do: :disabled
-
-   defp find_bundle(name) do
-     case Repo.get_by(Bundle, name: name) do
-       nil ->
-         {:error, {:not_found, name}}
-       %Bundle{}=bundle ->
-         {:ok, bundle}
-     end
-   end
-
-   defp error_message(:embedded_bundle),
-     do: "The status of the embedded bundle `#{Cog.embedded_bundle}` cannot be changed!"
-   defp error_message({:not_found, name}),
-     do: "The bundle `#{name}` cannot be found!"
-   defp error_message(:invalid_invocation),
-     do: "That is not a valid invocation of the `bundle` command"
-
+  def handle_message(%{args: ["status", bundle_name]} = req, state) do
+    case Bundles.with_status_by_name(bundle_name) do
+      nil ->
+        {:error, req.reply_to, error_message({:not_found, bundle_name}), state}
+      bundle ->
+        {:reply, req.reply_to, "bundle-status", bundle, state}
+    end
+  end
+
+  def handle_message(%{args: [action, "operable"|_]} = req, state)
+      when action in ["enable", "disable"] do
+    {:error, req.reply_to, error_message({:embedded_bundle, action}), state}
+  end
+
+  def handle_message(%{args: ["enable", bundle_name|args]} = req, state) do
+    result = with {:ok, bundle} <- parse_and_find_version(bundle_name, args),
+                  :ok           <- check_for_enabled_version(bundle_name),
+                  {:ok, bundle} <- enable_bundle(bundle),
+                  do: {:ok, bundle}
+
+    case result do
+      {:ok, bundle} ->
+        {:reply, req.reply_to, "bundle-enable", Map.put(bundle, :status, :enabled), state}
+      {:error, error} ->
+        {:error, req.reply_to, error_message(error), state}
+    end
+  end
+
+  def handle_message(%{args: ["disable", bundle_name]} = req, state) do
+    result = with {:ok, bundle} <- find_enabled_bundle(bundle_name),
+                  {:ok, bundle} <- disable_bundle(bundle),
+                  do: {:ok, bundle}
+
+    case result do
+      {:ok, bundle} ->
+        {:reply, req.reply_to, "bundle-disable", Map.put(bundle, :status, :disabled), state}
+      {:error, error} ->
+        {:error, req.reply_to, error_message(error), state}
+    end
+  end
+
+  def handle_message(req, state) do
+    {:error, req.reply_to, error_message(:invalid_invocation), state}
+  end
+
+  defp enable_bundle(%BundleVersion{}=bundle_version) do
```

Comment:
Since nothing is happening to the `bundle_version` you pass in (the same one is being passed back out), I think you can just call `Bundles.set_bundle_version_status/2` wherever you call this, but match on `:ok` instead.